### DISCUSSION TITLE
Option to disable morgan output in remix server

### DISCUSSION THREAD

**Discussion by kentcdodds** [remix-run/remix#2935] [Proposals]
The morgan output for the remix server makes it difficult for me to find my own logging output:

![image](https://user-images.githubusercontent.com/1500684/98396362-d766a900-201a-11eb-9b8f-f8b5b5e120df.png)

I already see all of these requests in my network tab. It's just noise for me. Could we get an option to disable the remix server morgan output?

I know that I could start the remix server in a separate tab or just silence the output, but then that would hide much-needed error messages.

**Comment by mjackson** [comment]
Could you just redirect stdout to `/dev/null` for now? You will still see error messages on stderr.

**Comment by kentcdodds** [comment]
Yup, that's a fine temporary workaround :) Thanks.

**Comment by cdock1029** [comment]
Good article describing some approaches for node / express logging to be configurable (express already setup this way) with environment var, and strategies of separation between app, library and cli logs.. as well as enabling configurable output based on environment..local/prod/CI 

https://www.twilio.com/blog/guide-node-js-logging

Turning on/off, filtering logs then can be done through

`DEBUG=express:*,remix:* yarn dev` 

The assorted `pino` libraries seem useful in this regard

**Comment by lumi-mustafa-hanif** [comment]
Really want this

**Reply by kiliman** [reply]
You'll need to either use the Express template (Remix App Server is basically a pre-configured Express server), or patch `@remix-run/serve` to remove the `morgan` call.
